# Programación en Python – variables

# Contenido

- [1. Guardando valores](#Guardando-valores)
- [2. Variables](#Variables)
- [3. Nombres de variables](#Nombres-de-variables)
- [4. Actualizando variables](#Actualizando-variables)
- [5. Atajos de sintaxis](#Atajos-de-sintaxis)

<a id="Guardando-valores"></a>
## 1. Guardando valores
[[Contenido]](#Contenido) 
[[Ejercicio de esta sección]](#Instrucciones-Guardar-valores)

En esta lección, aprenderemos cómo guardar valores en Python, cómo actualizar valores y cómo usar atajos de sintaxis para actualizar valores.

Digamos que queremos guardar el resultado de la operación aritmética `1.99 * 2` . A continuación se muestra el código que necesitamos ejecutar para guardar el resultado de esta multiplicación:

In [3]:
result = 1.99 * 2
print(result)

3.98


<a id="Instrucciones-Guardar-valores"></a>
## Instrucciones 
[[Contenido]](#Contenido) 
[[Teoria de esta sección]](#Guardando-valores)

<table>
<thead>
<tr>
<th>track_name</th>
<th>price</th>
<th>currency</th>
<th>rating_count_tot</th>
<th>user_rating</th>
</tr>
</thead>
<tbody>
<tr>
<td>Facebook</td>
<td>0.0</td>
<td>USD</td>
<td>2974676</td>
<td>3.5</td>
</tr>
<tr>
<td>Instagram</td>
<td>0.0</td>
<td>USD</td>
<td>2161558</td>
<td>4.5</td>
</tr>
<tr>
<td>Clash of Clans</td>
<td>0.0</td>
<td>USD</td>
<td>2130805</td>
<td>4.5</td>
</tr>
<tr>
<td>Fruit Ninja Classic</td>
<td>1.99</td>
<td>USD</td>
<td>698516</td>
<td>4.5</td>
</tr>
<tr>
<td>Minecraft: Pocket Edition</td>
<td>6.99</td>
<td>USD</td>
<td>522012</td>
<td>4.5</td>
</tr>
</tbody>
</table>

Con la información de la tabla de arriaba ¿Cuál sería el costo de comprar Minecraft: Pocket Edition para dos miembros de la familia?. Guarda el resultado de la operación en la variable `result` e imprimela.

Escribe tu codigo abajo de esta sección ⏬

---

In [4]:
# Ingresa aqui tu codigo


Escribe tu codigo arriba de esta sección ⏫

---

<a id="Variables"></a>
## 2. Variables 
[[Contenido]](#Contenido) 
[[Ejercicio de esta sección]](#Instrucciones-Variables)

Cuando ejecutamos el código anterior, 3.98 se almacena en una ubicación específica en la memoria de tu computadora. Cada ubicación de almacenamiento en la memoria de la computadora tiene una etiqueta o identificador único y, en este caso, la etiqueta es `result`. Esta etiqueta nos ayuda a encontrar y utilizar el valor almacenado (3.98) siempre que lo necesitemos.

Esta ubicación de almacenamiento etiquetada, `result`, es lo que llamamos **variable**. El nombre `result` es un nombre de variable.

El nombre `result` fue nuestra elección, pero podríamos haber usado cualquier otro nombre.

<a id="Instrucciones-Variables"></a>
## Instrucciones
[[Contenido]](#Contenido) 
[[Teoria de esta sección]](#Variables)

- Almacene el valor 6.99 en una variable llamada `minecraft_cost`.
- Almacene el resultado de 1.99 * 2 en una variable denominada `fruit_ninja_cost`.
- Usando la función `print()`, muestre lo siguiente:
    - El valor almacenado en la variable `minecraft_cost`.
    - El resultado de sumar 1.99 a la variable `fruit_ninja_cost`.
    - El resultado de agregar `fruit_ninja_cost` a `minecraft_cost`.

Escribe tu codigo abajo de esta sección ⏬

---

In [4]:
# Ingresa aqui tu codigo


Escribe tu codigo arriba de esta sección ⏫

---

<a id="Nombres-de-variables"></a>
## 3. Nombres de variables 
[[Contenido]](#Contenido) 
[[Ejercicio de esta sección]](#Instrucciones-Nombres-de-variables)


Estas son las dos reglas de sintaxis que debemos seguir cuando nombramos variables en Python:

1. Debemos utilizar sólo letras, números o guiones bajos (no podemos utilizar comas, comillas, guiones, espacios, etc.).
2. Los nombres de variables no pueden comenzar con un número.

En la siguiente tabla se muestran ejemplos de nombres de variables validas e invalidas.

<table>
  <tr>
    <th>Nombres Válidos</th>
    <th>Nombres Inválidos</th>
    <th>Razón de invalidez</th>
  </tr>
  <tr>
    <td>my_variable</td>
    <td>'invalid_name'</td>
    <td>Contiene comillas sencillas</td>
  </tr>
  <tr>
    <td>num_123</td>
    <td>123_num</td>
    <td>Comienza con un número</td>
  </tr>
  <tr>
    <td>some_name</td>
    <td>some name</td>
    <td>Contiene un espacio</td>
  </tr>
  <tr>
    <td>underscore_</td>
    <td>under,score</td>
    <td>Contiene una coma</td>
  </tr>
  <tr>
    <td>x</td>
    <td>-invalid_name</td>
    <td>Contiene un guion</td>
  </tr>
  <tr>
    <td>first_name</td>
    <td>first.name</td>
    <td>Contiene un punto</td>
  </tr>
</table>

<a id="Instrucciones-Nombres-de-variables"></a>
## Instrucciones
[[Contenido]](#Contenido) 
[[Teoria de esta sección]](#Nombres-de-variables)


Corrige el codigo que aparece debajo renombrando las varibles de manera correcta.

Escribe tu codigo abajo de esta sección ⏬

---

In [14]:
# Ingresa aqui tu codigo
facebook-rating = 3.5
instagram rating = 4.5

SyntaxError: cannot assign to expression here. Maybe you meant '==' instead of '='? (2634631567.py, line 2)

Escribe tu codigo arriba de esta sección ⏫

---

<a id="Actualizando-variables"></a>
## 4. Actualizando variables
[[Contenido]](#Contenido) 
[[Ejercicio de esta sección]](#Instrucciones-Actualizando-variables)

Podemos actualizar el valor almacenado en una variable. A continuación, primero almacenamos 1.99 en la variable `app_costs` y luego esta misma variable la actualizamos con un valor de 6.99.

In [16]:
app_costs = 1.99
print(app_costs)

app_costs = 6.99
print(app_costs)

1.99
6.99


Ahora analicemos el siguiente ejemplo de actualización de variables en Python.

In [ ]:
app_costs = 1.99
app_costs = app_costs + 6.99
print(app_costs)

Como puedes observar en el codigo de la celda de arriba cuando ejecutamos `app_costs = app_costs + 6.99`, `app_costs` se actualiza para almacenar el resultado de `app_costs + 6.99`, que es 8.98. Ejecutar `app_costs = app_costs + 6.99` es lo mismo que ejecutar `app_costs = 1.99 + 6.99` porque `app_costs` almacena 1.99.

<a name="Instrucciones-Actualizando-variables"></a>
## Instrucciones
[[Contenido]](#Contenido) 
[[Teoria de esta sección]](#Actualizando-variables)

<table>
<thead>
<tr>
<th>track_name</th>
<th>price</th>
<th>currency</th>
<th>rating_count_tot</th>
<th>user_rating</th>
</tr>
</thead>
<tbody>
<tr>
<td>Facebook</td>
<td>0.0</td>
<td>USD</td>
<td>2974676</td>
<td>3.5</td>
</tr>
<tr>
<td>Instagram</td>
<td>0.0</td>
<td>USD</td>
<td>2161558</td>
<td>4.5</td>
</tr>
<tr>
<td>Clash of Clans</td>
<td>0.0</td>
<td>USD</td>
<td>2130805</td>
<td>4.5</td>
</tr>
<tr>
<td>Fruit Ninja Classic</td>
<td>1.99</td>
<td>USD</td>
<td>698516</td>
<td>4.5</td>
</tr>
<tr>
<td>Minecraft: Pocket Edition</td>
<td>6.99</td>
<td>USD</td>
<td>522012</td>
<td>4.5</td>
</tr>
</tbody>
</table>



El recuento total de calificaciones (ver la tabla de arriba) de `Fruit Ninja Classic` es 698516, y el recuento total de calificaciones de `Minecraft: Pocket Edition` es 522012. Usemos estos valores en este ejercicio.

- Actualice la variable `rating_count_totals` agregando 522012 a su valor actual. La variable `rating_count_totals` ya está en la celda que aparece debajo.
- Imprimir `rating_count_totals`.

Escribe tu codigo abajo de esta sección ⏬

---

In [2]:
rating_count_totals = 698516
# Ingresa aqui tu codigo


Escribe tu codigo arriba de esta sección ⏫

---

<a id="Atajos-de-sintaxis"></a>
## 5. Atajos de sintaxis
[[Contenido]](#Contenido) 
[[Ejercicio de esta sección]](#Instrucciones-Atajos-de-sintaxis)

En el ejercicio anterior utilizamos el código `app_costs = app_costs + 6.99` para actualizar la variable `app_costs` de 1.99 a 8.98.

Hay varios atajos de sintaxis que podemos usar para actualizar una variable cuando realizamos operaciones aritméticas. Por ejemplo, como alternativa al código anterior, podemos escribir `app_costs += 6.99` (en lugar de `app_costs = app_costs + 6.99`):

In [3]:
app_costs = 1.99
app_costs += 6.99
print(app_costs)

8.98


A continuación se muestra una tabla con algunos atajos de sintaxis que podemos usar:


<table>
  <tr>
    <th>Atajo de Sintaxis</th>
    <th>Ejemplo de Código</th>
    <th>Resultado</th>
  </tr>
  <tr>
    <td>+=</td>
    <td>x += 5</td>
    <td>Valor de x incrementado en 5</td>
  </tr>
  <tr>
    <td>-=</td>
    <td>y -= 3</td>
    <td>Valor de y decrementado en 3</td>
  </tr>
  <tr>
    <td>*=</td>
    <td>z *= 2</td>
    <td>Valor de z multiplicado por 2</td>
  </tr>
  <tr>
    <td>/=</td>
    <td>a /= 4</td>
    <td>Valor de a dividido por 4</td>
  </tr>
  <tr>
    <td>**=</td>
    <td>b **= 3</td>
    <td>Valor de b elevado a la potencia de 3</td>
  </tr>
</table>

<a name="Instrucciones-Atajos-de-sintaxis"></a>
## Instrucciones
[[Contenido]](#Contenido) 
[[Teoria de esta sección]](#Atajos-de-sintaxis)

<table>
<thead>
<tr>
<th>track_name</th>
<th>price</th>
<th>currency</th>
<th>rating_count_tot</th>
<th>user_rating</th>
</tr>
</thead>
<tbody>
<tr>
<td>Facebook</td>
<td>0.0</td>
<td>USD</td>
<td>2974676</td>
<td>3.5</td>
</tr>
<tr>
<td>Instagram</td>
<td>0.0</td>
<td>USD</td>
<td>2161558</td>
<td>4.5</td>
</tr>
<tr>
<td>Clash of Clans</td>
<td>0.0</td>
<td>USD</td>
<td>2130805</td>
<td>4.5</td>
</tr>
<tr>
<td>Fruit Ninja Classic</td>
<td>1.99</td>
<td>USD</td>
<td>698516</td>
<td>4.5</td>
</tr>
<tr>
<td>Minecraft: Pocket Edition</td>
<td>6.99</td>
<td>USD</td>
<td>522012</td>
<td>4.5</td>
</tr>
</tbody>
</table>


Al observar la tabla de arriba te puedes dar cuenta que el recuento total de calificaciones de `Fruit Ninja Classic` es 698516, y el recuento total de calificaciones de `Minecraft: Pocket Edition` es 522012. El costo de `Fruit Ninja Classic` es 1.99. Usemos estos valores en este ejercicio.

- Asigna el valor de 698516 a una variable denominada `rating_count_totals`.
En este punto esta variable almacena el recuento total de valoraciones de `Fruit Ninja Classic`.

- Actualiza el valor de `rating_count_totals` agregando el recuento total de valoraciones de `Minecraft: Pocket Edition`. Para esto debes utilizar el operador +=.

- Ahora asigna el valor de 1.99 a una variable denominada `fruit_ninja_totals`. Esta variable almacena el costo de comprar `Fruit Ninja Classic`. Ahora actualiza esta variable para almacenar el precio de comprar esta aplicación para 4 personas. Actualiza el valor de `fruit_ninja_totals`  multiplicando su valor actual por 4 para lograrlo. Para esto debes utilizar el operador *=.

- Muestra `rating_count_totals` y `fruit_ninja_totals` usando `print()`.

Escribe tu codigo abajo de esta sección ⏬

---

In [1]:
# Ingresa aqui tu codigo


Escribe tu codigo arriba de esta sección ⏫

---